src/main/java/seedu/address/model/person/IncomeLevel.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's Income Level in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIncome(String)}
+ */
+public class IncomeLevel {
+
+    public static final String MESSAGE_CONSTRAINTS = "Invalid command format! \n"
+            + "add: Adds a person to the address book. Parameters: n/NAME p/PHONE e/EMAIL "
+            + "a/ADDRESS i/Income [t/TAG]...\n"
+            + "Example: add n/John Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, "
+            + "#02-25 i/$1000t/friends t/owesMoney";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "^(\\$)(\\d)+";
+
```

Comment:
Maybe can change the regex a bit because now adding leading zeroes will also be accepted. (e.g i/$000001,  i/$002)